wherehows-web/app/components/search/containers/search-facets.ts

```diff
@@ -0,0 +1,162 @@
+import { arrayMap } from 'wherehows-web/utils/array';
+import { IFacetsSelectionsMap, IFacetsCounts } from 'wherehows-web/utils/api/search';
+
+/**
+ * Options inside a facet
+ * @interface ISearchFacetOption
+ */
+interface ISearchFacetOption {
+  value: string;
+  label: string;
+}
+
+/**
+ * Interface of a facet
+ */
+interface ISearchFacet {
+  name: string;
+  displayName: string;
+  values: Array<ISearchFacetOption>;
+}
+
+/**
+ * Container component for search facets
+ * It will store state related to search facets.
+ */
+export default class SearchFacetsContainer extends Component {
+  didInsertElement(this: SearchFacetsContainer) {
+    get(this, 'getPlatformsTask').perform();
+  }
+
+  /**
+   * The current source to narrow search results to
+   * @type {DatasetPlatform}
+   */
+  currentSource: string;
+
+  /**
+   * Lists data platforms available to restrict search results by source
+   * @type {Array<DatasetPlatform>}
+   * @private
+   */
+  _sources: Array<DatasetPlatform> = [];
+
+  /**
+   * Current state of selections of facets
+   * EI:
+   * {
+   *  source: {
+   *    hdfs: true
+   *  },
+   *  fabric: {
+   *    corp: true,
+   *    prod: true
+   *  }
+   * }
+   */
+  selections: IFacetsSelectionsMap;
+
+  /**
+   * Counts for the facets in a similar fashion of selections
+   */
+  counts: IFacetsCounts;
+
+  /**
+   * Gets the available platforms and extracts a list of dataset sources
+   * @type {(Task<Promise<Array<IDataPlatform>>, (a?: any) => TaskInstance<Promise<Array<IDataPlatform>>>>)}
+   */
+  getPlatformsTask = task(function*(this: SearchFacetsContainer): IterableIterator<Promise<Array<IDataPlatform>>> {
+    const platforms: Array<IDataPlatform> = yield readPlatforms();
+    const getDatasetPlatform = ({ name }: IDataPlatform): DatasetPlatform => name;
+    const dataPlatforms = arrayMap(getDatasetPlatform)(platforms);
+
+    get(this, '_sources').setObjects(dataPlatforms);
+  });
+
+  /**
+   * I will convert a string into a facet option with counts
+   * @param facetValue
+   * @param facetName
+   */
+  stringToFacetOption(facetValue: string): ISearchFacetOption {
+    return {
+      value: facetValue,
+      label: capitalize(facetValue)
+    };
+  }
+
+  /**
+   * Creates a list of options with radio props for the data platforms that can be selected as a search filter
+   * @type {(ComputedProperty<Array<ISearchSourceOption>>}
+   */
+  @computed('_sources.[]', 'counts')
+  get sources(this: SearchFacetsContainer): Array<ISearchFacetOption> {
+    return this._sources.map(source => this.stringToFacetOption(source));
+  }
+
+  /**
+   * Facets that are available right now.
+   * In the future, it should be fetched from the backend
+   */
+  @computed('sources', 'counts')
+  get facets(): Array<ISearchFacet> {
+    return [
+      {
+        name: 'fabric',
+        displayName: 'Fabrics',
+        values: ['prod', 'corp', 'ei', 'dev'].map(fabric => this.stringToFacetOption(fabric))
```

Comment:
If this is to be fetched by the backend in teh future, would it be more prudent to set up this component to bind its data and be more separated and less "aware" of the mock data you're using?

Reply:
Well, this is not mock data. It is real data, but it will be potentially read from backend in the future. I guess I could put it in a constant, but I don't want to spread things around yet. I think all of this is going to change. There should be 1 single api to fetch all facet values, or maybe no api, since facets will come from search.